src/deps-index.ts

```diff
@@ -81,28 +29,176 @@ function _getHtmlDependencies(
  */
 export async function buildDepsIndex(
     entrypoints: ResolvedUrl[], analyzer: Analyzer): Promise<DepsIndex> {
-  const depsIndex = {
-    entrypointToDeps: new Map<ResolvedUrl, Set<ResolvedUrl>>()
-  };
+  const depsIndex = new Map<ResolvedUrl, Set<ResolvedUrl>>();
   const analysis = await analyzer.analyze(entrypoints);
   const allEntrypoints = new Set<ResolvedUrl>(entrypoints);
+  const inlineDocuments = new Map<ResolvedUrl, Document>();
 
   // Note: the following iteration takes place over a Set which may be added
   // to from within the loop.
   for (const entrypoint of allEntrypoints) {
     try {
-      const document = getAnalysisDocument(analysis, entrypoint);
-      const deps = getHtmlDependencies(document);
-      depsIndex.entrypointToDeps.set(
-          entrypoint, new Set([entrypoint, ...deps.eagerDeps]));
+      const document = inlineDocuments.has(entrypoint) ?
+          inlineDocuments.get(entrypoint)! :
+          getAnalysisDocument(analysis, entrypoint);
+      const deps = getDependencies(analyzer, document);
+      depsIndex.set(entrypoint, new Set([
+                      ...(document.isInline ? [] : [document.url]),
+                      ...deps.eagerDeps
+                    ]));
+
       // Add lazy imports to the set of all entrypoints, which supports
       // recursive
       for (const dep of deps.lazyImports) {
         allEntrypoints.add(dep);
       }
+
+      // Treat top-level script imports as entrypoints by creating "sub-bundle"
```

Comment:
Consider:
```html
<!-- document-a.html --->
<script src="./module-x.js" type="module"></script>
<script src="./module-y.js" type="module"></script>
```

If module-x and module-y both import module-z then we can't make module-z part of document-a's bundle.  However if ONLY module-x imports module-z than we CAN make module-z part of document-a's bundle because it can rollup into module-x when we inline module-x into document-a.

SO SIMPLE RIGHT?!?!